src/index.ts

```diff
@@ -156,99 +242,93 @@ export const findNextItem = (stimuli: Stimulus[], theta = 0, method = 'MFI', dee
       nextStimulus: stimuliAddFisher[0],
       remainingStimuli: stimuliAddFisher.slice(1),
     };
-  } else if (method === 'middle') {
+  }
+
+  private selectorMiddle(arr: Stimulus[]){
     let index: number;
-    if (arr.length < 5) {
+    if (arr.length < this.nStartItems) {
       index = Math.floor(arr.length / 2);
     } else {
-      index = Math.floor(arr.length / 2) + randomInteger(-2, 2);
+      index = Math.floor(arr.length / 2) + Cat.randomInteger(-Math.floor(this.nStartItems/2), Math.floor(this.nStartItems/2));
     }
     const nextItem = arr[index];
     arr.splice(index, 1);
     return {
       nextStimulus: nextItem,
       remainingStimuli: arr,
     };
-  } else if (method === 'closest') {
+  }
+
+  private selectorClosest(arr: Stimulus[]){
     //findClosest requires arr is sorted by difficulty
-    const index = findClosest(arr, theta + 0.481);
+    const index = Cat.findClosest(arr, this._theta + 0.481);
     const nextItem = arr[index];
     arr.splice(index, 1);
     return {
       nextStimulus: nextItem,
       remainingStimuli: arr,
     };
-  } else if (method === 'random') {
-    const index = Math.floor(Math.random() * arr.length);
+  }
+
+  private selectorRandom(arr: Stimulus[]){
+
+    const index = Math.floor(this._rng() * arr.length);
     const nextItem = arr.splice(index, 1)[0];
     return {
       nextStimulus: nextItem,
       remainingStimuli: arr,
     };
   }
 
-  /**
-   * return a random integer between min and max
-   * @param min - The minimum of the random number range (include)
-   * @param max - The maximum of the random number range (include)
-   * @returns {number} - random integer within the range
-   */
-  function randomInteger(min: number, max: number) {
-    return Math.floor(Math.random() * (max - min + 1)) + min;
-  }
+    /**
+     * return a random integer between min and max
+     * @param min - The minimum of the random number range (include)
+     * @param max - The maximum of the random number range (include)
+     * @returns {number} - random integer within the range
+     */
+    private static randomInteger(min: number, max: number) {
+      return Math.floor(Math.random() * (max - min + 1)) + min;
+    }
```

Comment:
Do you want to use `seedrandom` instead of `Math.random()` as you did in the `selectorRandom` method? I'm thinking the answer is yes so that you could make `selectorMiddle` replicable. But I'm certainly open to a differing view on this. If so, consider whether you want it to share the same `this._rng` that `selectorRandom` uses or whether you want it to have its own dedicated PRNG. I'd vote for sharing the same random state between the two methods.